### PR TITLE
DOCS: Developer docs for the hms-compcredentials package.

### PR DESCRIPTION
**Pull request by mpkelly-hpe**
Added developer docs for the hms-compcredentials package.  No other
changes were made, thus no version bump, etc.

### CODE REVIEW

**Review by nieuwsma — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+The keys used are typically component XNames.  The values stored are
+JSON-encoded data structures containing various types of sensitive information,
+including admin account usernames and passwords.
+
+
+## Data Types
+
+There are two main data types used with this package; one is the object 
+containing the plubming information for the back-end storage, and the other
```
Comment (mjendrysik-hpe) [inline README.md:17]: spelling error: plubming

File: README.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+group sets of key/value objects together.   The CompCreds package also provides
+this.  Each CompCredStore "handle" is opened/initialized with a key space 
+specified by the caller.  All calls using that handle will store and retrieve
+objects within that key space only.   If multiple key spaces are needed,
+multiple CompCredStore handles will be needed.
+
+
+## Protecting Sensitive Data
+
+Theh CompCredStore interface will always hide sensitive info when asked to
```
Comment (mjendrysik-hpe) [inline README.md:58]: Spelling error: Theh

**Review by mjendrysik-hpe — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (mpkelly-hpe) [reply README.md]: Fixed.

File: README.md (hunk 2 above)
Comment (mpkelly-hpe) [reply README.md]: Fixed.

**Pull request merged**